python/orca/src/bigdl/orca/common.py

```diff
@@ -183,79 +182,93 @@ def init_orca_context(cluster_mode="local", cores=2, memory="2g", num_nodes=1,
-                                executor_memory=memory, **spark_args)
-        else:
-            sc = init_spark_on_yarn(hadoop_conf=hadoop_conf,
-                                conda_name=conda_env_name,
-                                num_executors=num_nodes, executor_cores=cores,
-                                executor_memory=memory, **spark_args)
-    elif cluster_mode.startswith("k8s"):  # k8s or k8s-client
-        if cluster_mode == "k8s-cluster":
-            raise ValueError('For k8s-cluster mode, please set cluster_mode to "spark-submit" '
-                             'and submit the application via spark-submit instead')
-        assert "master" in kwargs, "Please specify master for k8s-client mode"
-        assert "container_image" in kwargs, "Please specify container_image for k8s-client mode"
-        for key in ["driver_cores", "driver_memory", "extra_executor_memory_for_ray",
-                    "extra_python_lib", "jars", "python_location"]:
-            if key in kwargs:
-                spark_args[key] = kwargs[key]
-        from bigdl.dllib.nncontext import init_spark_on_k8s
-        sc = init_spark_on_k8s(master=kwargs["master"],
-                               container_image=kwargs["container_image"],
-                               num_executors=num_nodes, executor_cores=cores,
-                               executor_memory=memory, **spark_args)
-    elif cluster_mode == "standalone":
-        for key in ["driver_cores", "driver_memory", "extra_executor_memory_for_ray",
-                    "extra_python_lib", "jars", "master", "python_location", "enable_numa_binding"]:
-            if key in kwargs:
-                spark_args[key] = kwargs[key]
-        from bigdl.dllib.nncontext import init_spark_standalone
-        sc = init_spark_standalone(num_executors=num_nodes, executor_cores=cores,
-                                   executor_memory=memory, **spark_args)
+        sc = init_nncontext(conf=conf, spark_log_level="WARN", redirect_spark_log=True)
     else:
-        raise ValueError("cluster_mode can only be local, yarn-client, standalone or spark-submit, "
-                         "but got: %s".format(cluster_mode))
+        cluster_mode = "local" if cluster_mode is None else cluster_mode
+        if cluster_mode == "spark-submit":
+            raise ValueError("spark-submit has been deprecated")
+        elif cluster_mode == "local":
+            assert num_nodes == 1, "For Spark local mode, num_nodes should be 1"
+            os.environ["SPARK_DRIVER_MEMORY"] = memory
+            if "python_location" in kwargs:
+                spark_args["python_location"] = kwargs["python_location"]
+            from bigdl.dllib.nncontext import init_spark_on_local
+            sc = init_spark_on_local(cores, **spark_args) 
+        elif cluster_mode.startswith("yarn"):  # yarn or yarn-client
+            if cluster_mode == "yarn-cluster":
+                raise ValueError('For yarn-cluster mode, '
```

Comment:
`yarn-cluster` is supported by #3313

Reply:
will update in a following PR to combine with init nncontext.